components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+                        }
+                        String gatewayEndpoint = environment.getApiGatewayEndpoint();
+                        if (gatewayEndpoint != null && gatewayEndpoint.contains(",")) {
+                            endpoint = gatewayEndpoint.split(",")[0];
+                        } else {
+                            endpoint = gatewayEndpoint;
+                        }
+                    }
+                }
+                //String apiPath = APIUtil.getAPIPath(apiIdentifier);
+                if (endpoint.endsWith(RegistryConstants.PATH_SEPARATOR)) {
+                    endpoint.substring(0, endpoint.length() - 1);
+                }
+                // We do not need the version in the base path since with the context version strategy, the version is
+                // embedded in the context
+                String basePath = endpoint + api.getContext();
+                resourceConfig.put("basePath", basePath);
+
+                JSONArray resource_configs = (JSONArray) resourceConfig.get("apis");
+
+                //Iterating each Sub resourcePath config
+                int subResCount = 0;
+                while (subResCount < resource_configs.size()) {
+                    JSONObject subResource = (JSONObject) resource_configs.get(subResCount);
+                    String uriTempVal = (String) subResource.get("path");
+                    uriTempVal = uriTempVal.startsWith("/") ? uriTempVal : ("/" + uriTempVal);
+
+                    JSONArray operations = (JSONArray) subResource.get("operations");
+                    //Iterating each operation config
+                    for (Object operation1 : operations) {
+                        JSONObject operation = (JSONObject) operation1;
+                        String httpVerb = (String) operation.get("method");
+                        /* Right Now PATCH is not supported. Need to remove this check when PATCH is supported*/
+                        if (!"PATCH".equals(httpVerb)) {
+                            URITemplate template = new URITemplate();
+                            Scope scope = APIUtil.findScopeByKey(getScopes(resourceConfigsJSON), (String) operation.get("scope"));
+
+                            String authType = (String) operation.get("auth_type");
+                            if (authType != null) {
+                                if (authType.equals("Application & Application User")) {
+                                    authType = APIConstants.AUTH_APPLICATION_OR_USER_LEVEL_TOKEN;
+                                }
+                                if (authType.equals("Application User")) {
+                                    authType = "Application_User";
+                                }
+                            } else {
+                                authType = APIConstants.AUTH_NO_AUTHENTICATION;
+                            }
+                            template.setThrottlingTier((String) operation.get("throttling_tier"));
```

Comment:
Can use constants for the string literals used in this particular line as well as the one follows this too.